### PR TITLE
Push docs to separate repo

### PR DESCRIPTION
**Pull request by aboedo**
This separates the docs into a new repo, [`purchases-ios-docs`](https://github.com/revenuecat/purchases-ios-docs). Having it detached from the main repo means that we can easily have docs for multiple versions of the SDK live there without needing to worry about workflows (like for hotfixes). I already backfilled docs for 3.14.1, 4.0.0 and 4.1.0 (even 4.2.0-SNAPSHOT)

So this generates the docs as usual, but instead of adding them to the `docs` folder in this repo, it adds them to the other repo and pushes. After pushing they'll be automatically published. 

You can see this working in https://revenuecat.github.io/purchases-ios-docs. 

For different versions: 

- 4.2.0-SNAPSHOT: https://revenuecat.github.io/purchases-ios-docs/4.2.0-SNAPSHOT/documentation/revenuecat/
- 4.1.0: https://revenuecat.github.io/purchases-ios-docs/4.1.0/documentation/revenuecat/
- 4.0.0: https://revenuecat.github.io/purchases-ios-docs/4.0.0/documentation/revenuecat/
- 3.14.1: https://revenuecat.github.io/purchases-ios-docs/3.14.1

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -1,8 +0,0 @@
-<!DOCTYPE html>
```
Comment (aboedo) [inline docs/index.html:1]: this file was moved to scripts/docs/index.html. the docs folder is getting deleted entirely in a follow-up PR

File: fastlane/Fastfile [hunk 2]
```diff
@@ -447,28 +447,53 @@ platform :ios do
   lane :generate_docs do
```
Comment (aboedo) [inline fastlane/Fastfile:447]: let me know if the logic in this new version of the lane is hard to follow. 
it does: 
- doc generation
- clone the docs repo
- copy the stuff over to the cloned repo
- git add, commit, push

File: fastlane/Fastfile [hunk 3]
```diff
@@ -447,28 +447,53 @@ platform :ios do
...
-         "--target",
-         "RevenueCat",
-         "--disable-indexing",
-         "--output-path",
-         docs_folder,
-         "--hosting-base-path",
-         "#{hosting_base_path}",
-         "--transform-for-static-hosting")
+    hosting_base_path = "purchases-ios-docs/#{version_number}"
+    docs_repo_base_url = "https://github.com/revenuecat/"
```
Comment (NachoSoto) [inline fastlane/Fastfile:451]: Should this be `ssh` instead (`"git@github.com:RevenueCat/" + docs_repo_name`)

**Review by NachoSoto — APPROVED** [review]

File: fastlane/Fastfile (hunk 3 above)
Comment (aboedo) [reply fastlane/Fastfile:451]: that's a great idea! I'll try it out in a follow-up PR

Comment (joshdholtz) [reply fastlane/Fastfile:451]: Can we make this an environment variable and the default it to either `https://github.com/revenuecat/` or `git@github.com:RevenueCat/` if not set?

File: fastlane/Fastfile [hunk 4]
```diff
@@ -447,28 +447,53 @@ platform :ios do
...
+           "generate-documentation",
+           "--target",
+           "RevenueCat",
+           "--disable-indexing",
+           "--output-path",
+           docs_generation_folder,
+           "--hosting-base-path",
+           "#{hosting_base_path}",
+           "--transform-for-static-hosting")
+        docs_index_path = "#{Dir.pwd}/scripts/docs/index.html"
```
Comment (joshdholtz) [inline fastlane/Fastfile:473]:
```suggestion
        docs_index_path = File.join(Dir.pwd, "/scripts/docs/index.html")
```

File: fastlane/Fastfile [hunk 5]
```diff
@@ -447,28 +447,53 @@ platform :ios do
...
+              docs_destination_folder = "docs/#{version_number}"
+              index_destination_path = "docs/index.html"
+              FileUtils.mv docs_generation_folder, docs_destination_folder
+              FileUtils.cp docs_index_path, index_destination_path
+
+              # using sh instead of fastlane commands because fastlane would run
+              # from the repo root
+              sh("git", "add", docs_destination_folder)
+              sh("git", "add", index_destination_path)
+              sh("git", "commit", "-m", 'Update documentation')
```
Comment (joshdholtz) [inline fastlane/Fastfile:490]: Maybe put the version number in the commit?

```suggestion
              sh("git", "commit", "-m", "Update documentation for #{verison_number}")
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]

File: fastlane/Fastfile (hunk 5 above)
Comment (aboedo) [reply fastlane/Fastfile]: great call!

File: fastlane/Fastfile (hunk 4 above)
Comment (aboedo) [reply fastlane/Fastfile]: great call, done!

**Comment by aboedo** [comment]
ready for another pass

File: fastlane/Fastfile [hunk 6]
```diff
@@ -447,28 +447,53 @@ platform :ios do
...
-         "--target",
-         "RevenueCat",
-         "--disable-indexing",
-         "--output-path",
-         docs_folder,
-         "--hosting-base-path",
-         "#{hosting_base_path}",
-         "--transform-for-static-hosting")
+    docs_repo_base_url = ENV["DOCS_REPO_BASE_URL"]
+    docs_repo_name = ENV["DOCS_REPO_NAME"]
```
Comment (joshdholtz) [inline fastlane/Fastfile:451]: I also might add something like...

```rb
UI.user_error!("Missing environment variable: DOCS_REPO_BASE_URL") unless docs_repo_base_url
UI.user_error!("Missing environment variable: DOCS_REPO_NAME") unless docs_repo_name
``

Just to prevent some unknown error confusion if these aren't set 🤷‍♂️

**Review by joshdholtz — CHANGES_REQUESTED** [review]
One more small thing!

File: fastlane/Fastfile (hunk 6 above)
Comment (aboedo) [reply fastlane/Fastfile:451]: love it, adding

**Review by joshdholtz — APPROVED** [review]
💯 LGTM
